src/services/checkSavedApplicationService.ts

```diff
@@ -1,25 +1,33 @@
 import { Response } from "express";
 import Resource from "@companieshouse/api-sdk-node/dist/services/resource";
-import { TransactionList } from "@companieshouse/api-sdk-node/dist/services/transaction/types";
+import { TransactionData, TransactionList } from "@companieshouse/api-sdk-node/dist/services/transaction/types";
 import { Session } from "@companieshouse/node-session-handler";
-import { APPROVED, CLOSED, REJECTED } from "../common/__utils/constants";
+import { APPROVED, CLOSED, REJECTED, RESUME_APPLICATION_ID } from "../common/__utils/constants";
 import { BASE_URL, CANNOT_REGISTER_AGAIN, CANNOT_SUBMIT_ANOTHER_APPLICATION, CHECK_SAVED_APPLICATION, SAVED_APPLICATION, TYPE_OF_BUSINESS } from "../types/pageURL";
 import logger from "../utils/logger";
 import { deleteAcspApplication } from "./acspRegistrationService";
 import { ErrorService } from "./errorService";
 
 export const getRedirectionUrl = async (transactionlistResource: Resource<TransactionList>, session: Session, res: Response, locales:any, lang:string): Promise<string> => {
     const transactionList = transactionlistResource.resource;
-    const transaction = transactionList?.items[transactionList?.items.length - 1];
+    const length = transactionList!.items.length;
+    for (let i = 0; i < length; i++) {
+        logger.debug("element------>" + transactionList!.items[i].id);
```

Comment:
Do we need this logger statement?

Reply:
It's not needed. Will keep it until testing is finished. Amrita complains that instead of latest transaction, the old one is getting picked up.Will remove this once testing is finished.